redis/redis.go

```diff
@@ -175,7 +175,41 @@ func (r *Redis) DeleteKeyMatch(match string) (int, error) {
 	return r.DeleteKeyMatchFn(match, nil)
 }
 
+func (r *Redis) ScanAll(match string) ([]string, error) {
```

Comment:
Oh, I actually wanted to avoid implementing something like this. I'd prefer to refactor the DeleteMatch method into 2: one method for Iteration with scanning (similar to this only not chewing memory) and DeleteMatch utilizes this. Call this method `ScanIterate(match string, fn func(key string)`. I wanted to avoid implementing a method that get's all to avoid programming without being conscious of memory limitations.